cubes/new_cube.py

```diff
@@ -0,0 +1,254 @@
+""" JXP VERSION """
+
+'''
+Questions for Clancy:
+
+1. lC vs. C
+2. Order of update_grid()
+3. Survey is unique to a grid, right?
+4. Order of grid parameters
+5. Why 90s per iteration?
```

Comment:
When testing this, the main conclusion was that we are performing calculations over large grids of O~1 million points. I tried speeding this up with... I forget which pre-compiled c library code I tried - and it didn't help, the conclusion being that numpy is highly optimised already for large grid operations.